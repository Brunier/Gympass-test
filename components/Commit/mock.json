{
  "sha": "942a765b53662936679b8119ed4533d972298d2a",
  "commit": {
    "author": {
      "name": "Pedro Bullo Junior",
      "email": "pedro.bj@hotmail.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "committer": {
      "name": "GitHub",
      "email": "noreply@github.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "message": "Workaround empty child",
    "tree": {
      "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
      "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
    },
    "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
    "comment_count": 0,
    "verification": {
      "verified": true,
      "reason": "valid"
    }
  },
  "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
}
