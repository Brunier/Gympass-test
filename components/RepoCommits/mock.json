{
  "data" : [
    {
      "sha": "942a765b53662936679b8119ed4533d972298d2a",
      "commit": {
        "author": {
          "name": "Pedro Bullo Junior",
          "email": "pedro.bj@hotmail.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "message": "Workaround empty child",
        "tree": {
          "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
    },
    {
      "sha": "c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
      "commit": {
        "author": {
          "name": "matc4",
          "email": "matias.s.cortes@gmail.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "message": "Merge pull request #100 from nesc58/masteFix: Text elements / transform + clipPath attribute added",
        "tree": {
          "sha": "97dae56ead52ab59e12b98848f4e38d89265040f",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/97dae56ead52ab59e12b98848f4e38d89265040f"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c"
    }
  ]
}
