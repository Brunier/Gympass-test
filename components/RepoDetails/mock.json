{
  "name": "interview-test",
  "description": "Interview test project",
  "url": "https://api.github.com/repos/Gympass/interview-test",
  "commits_url": "https://api.github.com/repos/Gympass/interview-test/commits",
  "created_at": "2018-05-22T19:57:37Z",
  "updated_at": "2019-01-24T02:51:23Z",
  "language": "Javascript",
  "stars": 3,
  "watchers": 3,
  "forks": 4
}
