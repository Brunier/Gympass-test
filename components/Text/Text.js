import styled from 'styled-components';
import React from 'react';



let Text = styled.span`
  color: #585656;
  font-size: ${props => props.fontSize ? props.fontSize + "px"  : "20px"};
  font-weight: ${props => props.bold ? "bold" : "none"};
`

export default Text;

