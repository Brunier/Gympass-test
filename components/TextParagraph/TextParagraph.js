import styled from 'styled-components';
import React from 'react';


let TextParagraph = styled.p`
  overflow: hidden;
  position: relative;
  line-height: 16px;
  max-height: ${props => props.maxHeight ? props.maxHeight + "px"  : "96px"}; 
  height: ${props => props.height ? props.height + "px" : "inherit"}; 
  margin-right: -1em;
  padding-right: 1em;
  color: #585656;
`

export default TextParagraph;


/*
 color: #585656;
  font-size: 16px;
    height: 96px;
    overflow: hidden;
    text-overflow: ellipsis;
    max-height: 90px;
 */
