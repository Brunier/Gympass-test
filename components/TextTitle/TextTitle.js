import styled from 'styled-components';
import React from 'react';

let TextTitle = styled.h1`
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: ${props => props.maxWidth ? props.maxWidth : "90%"};
    display: ${props => props.display ? props.display : "block"};
    font-size: ${props => props.fontSize ? props.fontSize + "px" : "20px"};
    color: #fff;
    padding: ${props => props.padding ? props.padding : "5px"};
    background: #00757e;
    margin-top: 0 
`

export default TextTitle;

