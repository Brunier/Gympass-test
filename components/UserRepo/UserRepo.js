import React from 'react';
import styled from "styled-components";
import Text from "../Text/Text";
import TextParagraph from "../TextParagraph/TextParagraph";
import TextTitle from "../TextTitle/TextTitle";
import moment from 'moment'
import momentBr from 'moment/locale/pt-br'
moment.locale('momentBr');
const Container = styled.div`
  width: 21%;
  border: 0.13vw solid #00757e;
  margin-right: 2%;
  padding: 1%;
  margin-bottom: 20px;
  cursor: pointer;
  
  &:hover {
    background: #00757e;
    
    span, p {
      color: #fff;
    }
  }
  
  &:nth-child(4n) {
      margin-right: 0 
    }
`;

 class UserRepo extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <Container onClick={this.props.onClick}>
                <TextTitle padding={"5px 10% 5px 5px"}>{this.props.repo.name}</TextTitle>
                <TextParagraph height={96}>{this.props.repo.description}</TextParagraph>
                <Text bold={true} fontSize={14}>{moment(this.props.repo.updated_at).format('llll')}</Text>
                <Text bold={true} fontSize={14}> - </Text>
                <Text bold={true} fontSize={14}>{this.props.repo.language ? this.props.repo.language : "Nenhuma"}</Text>

            </Container>
        )
    }
}

export default UserRepo;
