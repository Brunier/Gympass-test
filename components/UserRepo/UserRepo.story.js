import React from 'react';
import { storiesOf } from '@storybook/react';
import UserRepo from './UserRepo';
import mock from "./mock"


storiesOf('UserRepo', module)
    .add('Normal', () => (
        <UserRepo  repo={mock}/>
    ))
