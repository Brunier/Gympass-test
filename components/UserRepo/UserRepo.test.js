import React from 'react';
import { shallow } from 'enzyme';
import TextTitle from '../TextTitle/TextTitle';
import TextParagraph from '../TextParagraph/TextParagraph';
import Text from '../Text/Text';
import renderer from 'react-test-renderer'
import mock from './mock'
// Components
import UserRepo from './UserRepo';
import moment from "moment/moment";

function setup(props) {
    const wrapper = shallow(<UserRepo {...props} />);
    return { wrapper };
}

const repo = mock;

describe('UserSearch Test Suite', () => {
    it('Titulo', () => {
        const { wrapper } = setup({repo});
        expect(wrapper.find(TextTitle).text()).toBe(repo.name);
    });
    it('Descrição', () => {
        const { wrapper } = setup({repo});
        expect(wrapper.find(TextParagraph).text()).toBe(repo.description);
    });
    it('Data de atualização', () => {
        const { wrapper } = setup({repo});
        expect(wrapper.find(Text).at(0).text().trim()).toBe(moment(repo.updated_at).format('llll'));
    });
    it('Linguagem', () => {
        const { wrapper } = setup({repo});
        expect(wrapper.find(Text).at(2).text()).toBe(repo.language);
    });
    it('Snapshot', () => {
        const tree = renderer.create(<UserRepo repo={repo} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
