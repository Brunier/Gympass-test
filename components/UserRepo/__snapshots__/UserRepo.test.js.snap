// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Test Suite Snapshot 1`] = `
<div
  className="sc-bxivhb jmszxS"
  data-cy="repo"
>
  <h1
    className="sc-bdVaJa eiPDuz"
    data-cy="repo-title"
  >
    interview-test
  </h1>
  <p
    className="sc-bwzfXH fRwYcu"
    height={96}
  >
    Interview test project
  </p>
  <span
    className="sc-htpNat ivUPnB"
    fontSize={14}
  >
    Qui, 24 de Jan de 2019 Ã s 00:51
  </span>
  <span
    className="sc-htpNat ivUPnB"
    fontSize={14}
  >
     - 
  </span>
  <span
    className="sc-htpNat ivUPnB"
    fontSize={14}
  >
    Javascript
  </span>
</div>
`;
